app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt

```diff
@@ -35,8 +35,15 @@ class DownloadFragmentStore(initialState: DownloadFragmentState) :
 /**
  * Actions to dispatch through the `DownloadStore` to modify `DownloadState` through the reducer.
  */
+
 sealed class DownloadFragmentAction : Action {
     object ExitEditMode : DownloadFragmentAction()
+    data class AddItemForRemoval(val item: DownloadItem) : DownloadFragmentAction()
+    data class RemoveItemForRemoval(val item: DownloadItem) : DownloadFragmentAction()
+    data class AddPendingDeletionSet(val itemIds: Set<String>) : DownloadFragmentAction()
+    data class UndoPendingDeletionSet(val itemIds: Set<String>) : DownloadFragmentAction()
+    object EnterDeletionMode : DownloadFragmentAction()
+    object ExitDeletionMode : DownloadFragmentAction()
```

Comment:
Could it be useful to document each of the actions? I'm a bit confused between `AddItemForRemoval` vs `AddPendingDeletionSet`?

Reply:
This is where I would defer to @boek for advice because this is directly based on HistoryFragment's implementation

Reply:
So one is for the UI to determine what is selected for deletion to toggle like the checkmark, the other is for the undo logic. I can definitely add some comments around that!

Reply:
Thanks for the clarification 👍🏽